src/main/scala/Robot.scala

```diff
@@ -8,58 +8,68 @@ object Directions {
   object Right extends Directions
 }
 
-class Robot {
-  var coordinates: (Int, Int) = (0, 0)
+case class Coordinates(x:Int = 0, y:Int = 0)
 
+class Robot {
   private def parseInstructions(instructions: String): Seq[Directions] = {
     def parseInstruction(char: Char): Option[Directions] =
       char match {
         case 'U' => Some(Directions.Up)
         case 'D' => Some(Directions.Down)
         case 'L' => Some(Directions.Left)
         case 'R' => Some(Directions.Right)
-        case _ => None
+        case _   => None
       }
-    instructions.flatMap(instruction => parseInstruction(instruction))
+    instructions.flatMap(parseInstruction)
+  }
+
+  private def moveAll(coordinates: Coordinates, instructions: Seq[Directions]): Coordinates = {
+    if (instructions.isEmpty)
+      coordinates
+    else
+      moveAll(moveSingle(coordinates, instructions.head), instructions.tail)
   }
 
-  private def moveSingle(instruction: Directions): (Int, Int) = {
+  private def moveSingle(coordinates: Coordinates, instruction: Directions): Coordinates = {
     instruction match {
-      case Directions.Up => moveY(1)
-      case Directions.Down => moveY(-1)
-      case Directions.Left => moveX(-1)
-      case Directions.Right => moveX(1)
+      case Directions.Up    => moveY(coordinates, 1)
+      case Directions.Down  => moveY(coordinates, -1)
+      case Directions.Left  => moveX(coordinates, -1)
+      case Directions.Right => moveX(coordinates, 1)
     }
   }
 
-  private def moveX(amount: Int): (Int, Int) = {
-    val (x, y) = this.coordinates
-    (checkBounds(x + amount), y)
+  private def moveX(coordinates: Coordinates, amount: Int): Coordinates = {
+    coordinates.copy(
+      checkBounds(coordinates.x + amount),
+      coordinates.y
+    )
   }
 
-  private def moveY(amount: Int): (Int, Int) = {
-    val (x, y) = this.coordinates
-    (x, checkBounds(y + amount))
+  private def moveY(coordinates: Coordinates, amount: Int): Coordinates = {
+    coordinates.copy(
+      coordinates.x,
+      checkBounds(coordinates.y + amount)
+    )
   }
 
   private def checkBounds(coordinate: Int): Int = {
     val BOUNDS = 10
+    def wrap = (bound: Int) => bound * 2 + 1
+
     if (coordinate > BOUNDS)
-      coordinate - (BOUNDS * 2 + 1)
+      coordinate - wrap(BOUNDS)
     else if (coordinate < -BOUNDS)
-      coordinate + (BOUNDS * 2 + 1)
+      coordinate + wrap(BOUNDS)
     else
       coordinate
   }
 
   def move(instructions: String): (Int, Int) = {
     val parsedInstructions: Seq[Directions] = parseInstructions(instructions)
+    val startingCoordinates = Coordinates(0,0)
```

Comment:
Generally, put magic numbers (like these `startingCoordinates`) that occur inside functions or classes as part of their parameterization, but as default values.

A pattern I like to have in my smaller programs is to have an object at the top-level of the program called `Defaults`, then I use that object like: `move(..., origin = Defaults.origin)`.  It makes it easy to extract pieces of `Defaults` out as sensible collections of traits and fulfill them with configuration-hydrated objects to wire in as they need to become configurable (and sometimes they don't, but it at least organizes it.)

Reply:
Good advice. Moved the two magic values to a `Defaults` object